### PR TITLE
Update Ecosia search engine params

### PR DESCRIPTION
**Pull request by bsclifton**
Fixes https://github.com/brave/brave-browser/issues/16639

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used GitHub [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
See https://github.com/brave/brave-browser/issues/16639

### CODE REVIEW

**Review by mkarolin — APPROVED** [review]
LGTM

File: components/search_engines/brave_prepopulated_engines.cc [hunk 1]
```diff
@@ -131,7 +131,7 @@ const PrepopulatedEngine brave_ecosia =
 #else
                        "e8eb07a6"
 #endif
-                       "&q={searchTerms}",
+                       "&q={searchTerms}&addon=brave",
```
Comment (simonhong) [inline components/search_engines/brave_prepopulated_engines.cc:134]: Just curious how this change affetcs ecosia page?

**Comment by simonhong** [comment]
With this change, ecosia.org always hide `Add to Chrome` button even if ecosia is not a default se.
This issue description says `This shouldn't be showing for Brave users who have Ecosia set as their default search engine`.
This issue description needs to be updated or this is just side effect?

**Comment by bsclifton** [comment]
Good point @simonhong - let me update the issue. I'll raise with folks at Ecosia too 😄

File: components/search_engines/brave_prepopulated_engines.cc (hunk 1 above)
Comment (simonhong) [reply components/search_engines/brave_prepopulated_engines.cc:134]: @bsclifton Ah,, ecosia page can read this string?

Comment (bsclifton) [reply components/search_engines/brave_prepopulated_engines.cc:134]: Good question; I'm not exactly sure - I didn't notice anything special in local storage or cookies. Change is coming at request of the team.

**Review by simonhong — APPROVED** [review]

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 10 x64` using the following build:

```
Brave | 1.28.37 Chromium: 91.0.4472.124 (Official Build) nightly (64-bit)
--- | ---
Revision | 7345a6d1bfcaff81162a957e9b7d52649fe2ac38-refs/branch-heads/4472_114@{#6}
OS | Windows 10 OS Version 2009 (Build 19042.1081)
```

Reproduced the original issue using `1.28.35 Chromium: 91.0.4472.124` as per the following:

**`Example`** | **`Example`** | **`Example`**
----------------|---------------|----------------
ecosia2 | ecosia1 | ecosia3

When using `1.28.35 Chromium: 91.0.4472.124`, you'll notice that `https://www.ecosia.org/search?tt=e8eb07a6&q=` is being used without the addition of `&addon=brave` which was added in this PR.

Verified the STR/Cases via https://github.com/brave/brave-browser/issues/16639#issue-931298170 with `1.28.37 Chromium: 91.0.4472.124` as per the following:

## `Test Case #1`  (Clean install)

* ensured that visiting https://www.ecosia.org displayed the `Add Ecosia to Chrome` on a clean profile
  * Ecosia wasn't set as the default and a search was never completed
* ensured that the `Add Ecosia to Chrome` button wasn't being displayed when setting Ecosia as default and completing a search
* ensured that `addon=brave` was being added into the search query

**`Example`** | **`Example`**
----------------|---------------
ecosiaLatest1 | ecosiaLatest2

**Comment by GeetaSarvadnya** [comment]
Verification is in-progress

```

Brave | 1.26.74 Chromium: 91.0.4472.124 (Official Build) (64-bit)
-- | --
Revision | 7345a6d1bfcaff81162a957e9b7d52649fe2ac38-refs/branch-heads/4472_114@{#6}
OS | Windows 10 OS Version 2004 (Build 19041.1052)


```

**Pull request merged**